File: SpendWiseLoginWebApi/LoginRestApi/Models/Orgcategory.cs
using System;
using System.Collections.Generic;

namespace LoginRestApi.Models;

public partial class Orgcategory
{
    public int Orgcid { get; set; }

    public string Orgcname { get; set; } = null!;

    public string Cdesc { get; set; } = null!;

    public virtual ICollection<Orgexpenditure> Orgexpenditures { get; set; } = new List<Orgexpenditure>();
}

File: SpendWiseRestApi/src/main/java/com/example/demo/entity/OrgCategory.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Entity
@Table(name = "orgcategory")
@JsonIgnoreProperties({"orgExpenditures"})
public class OrgCategory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "orgcid")
    private int orgcId;

    @Column(name = "orgcname", unique = true, nullable = false)
    private String orgcName;

    @Column(name = "cdesc", nullable = false)
    private String cDesc;

    @OneToMany(mappedBy = "orgCategory", cascade = CascadeType.ALL)
    @JsonIgnoreProperties("orgCategory")
    private Set<OrgExpenditure> orgExpenditures;
}
